core/Binary.carp

```diff
@@ -0,0 +1,111 @@
+(system-include "carp_binary.h")
+(load "StdInt.carp")
+
+(defmodule Binary
+  (doc Order 
+    "The type of byte orders. 
+
+    LittleEndian designates the little endian ordering, and indicates the least
+    significant byte appears first in a given byte sequence.
+
+    BigEndian designates the big endian ordering, and indicates the most
+    significant byte occurs first in a given byte sequence.")
+  (deftype Order LittleEndian BigEndian)
+
+  (register to-int16 (λ [Byte Byte] Uint16))
+  (register to-int32 (λ [Byte Byte Byte Byte] Uint32))
+  (register to-int64 (λ [Byte Byte Byte Byte Byte Byte Byte Byte] Uint64))
+  (register system-endianness-internal (λ [] Int))
+
+  (doc system-endianness 
+    "Returns the endianness of the host system.")
+  (sig system-endianness (λ [] Order))
+  (defn system-endianness [] 
+    (if (= (system-endianness-internal) 1)
+        (Order.LittleEndian)
+        (Order.BigEndian)))
+
+  (doc bytes->int16-unsafe
```

Comment:
Should we also introduce safe variants of those functions?

Reply:
Sorry, didn’t remember that this is actually already in the todo list.

Reply:
No prob! They're in :)